Removed "Other Subjects", modified listed Subjects

daniel-li95 opened the pull request:
Removed Other Subjects text box input, Modified listed subjects to
remove "other" and added an optional What we learned field using the
most applicable metadata element.

Also changed the tab spacing in messages.properties (jsp.about.title) to
spaces.

daniel-li95 committed:
Removed "Other Subjects", modified listed Subjects

Removed Other Subjects text box input, Modified listed subjects to
remove "other" and added an optional What we learned field using the
most applicable metadata element.

Also changed the tab spacing in messages.properties (jsp.about.title) to
spaces.

daniel-li95 committed:
Replaced all subjects with psuedo-tree structure

Removed all other input forms in input-forms.xml and replaced all value
pairs with the relevant pairs.

ionparticle commented on dspace/config/input-forms.xml:
```diff
@@ -5884,794 +199,558 @@
-     </pair>
+    <pairs>
+	   <displayed-value>Combinatorics >>> Counting >>> Combinations</displayed-value>
+	   <stored-value>Combinatorics >>> Counting >>> Combinations</stored-value>
```
Looks like stored value shouldn't have special characters, so let's replace the spaces and greater than signs with underscores. E.g.: Combinatorics_____Counting_____Combinations

daniel-li95 committed:
Replaced all whitespace in stored-value
Replaced all whitespace between stored-value tags with underscores(_)
and all >>> with triple underscores(___)
Other special characters such as hyphen (-) and slash (/) still exist

daniel-li95 committed:
Corrected input-forms.xml

Corrected the word from pairs to pair and readded the missing resources
value-pairs. Tested to show that the wording is correct and displays
correctly on the site.
